tests/ClientBuilderTest.php

```diff
@@ -210,6 +210,38 @@ public function optionsDataProvider()
             ['setErrorTypes', 0],
         ];
     }
+
+    public function testCompressionPluginAdded()
+    {
+        $builder = ClientBuilder::create(['enableCompression' => true, 'dsn' => 'https://completelyrandom@dsn.asdf/42']);
+        $builder->getClient();
+        $reflectionProperty = new \ReflectionProperty(ClientBuilder::class, 'httpClientPlugins');
+        $reflectionProperty->setAccessible(true);
+        $foundDecoderPlugin = false;
+        foreach ($reflectionProperty->getValue($builder) as $plugin) {
+            if ($plugin instanceof Plugin\DecoderPlugin) {
+                $foundDecoderPlugin = true;
+            }
+        }
+        $this->assertTrue($foundDecoderPlugin);
+        $reflectionProperty->setAccessible(false);
+    }
+
+    public function testCompressionPluginNotAdded()
```

Comment:
If we use a data provider for the test above we could just accept as parameter of the test if we expect or not the plugin instance to exists in the list of plugins and then assert against that